certbot/display/ops.py

```diff
@@ -289,3 +289,49 @@ def _gen_https_names(domains):
             domains[-1])
 
     return ""
+
+
+def _get_validated(method, validator, message, default=None, **kwargs):
+    if default is not None:
+        try:
+            validator(default)
+        except errors.PluginError as error:
+            raise AssertionError("Invalid default {0}".format(default))
+
+    while True:
+        code, raw = method(message, default=default, **kwargs)
+        if code == display_util.OK:
+            try:
+                validator(raw)
+                return code, raw
+            except errors.PluginError as error:
+                z_util(interfaces.IDisplay).notification(str(error), pause=False)
+        else:
+            return code, raw
+
+
+def validated_input(validator, *args, **kwargs):
+    """Like ~certbot.interfaces.IDisplay.input, but with validation.
+    
+    :param validator: 
+    :param list *args: Arguments to be passed to ~certbot.interfaces.IDisplay.input
+    :param dict **kwargs: Arguments to be passed to ~certbot.interfaces.IDisplay.input 
+    :return: as ~certbot.interfaces.IDisplay.input
+    :rtype: tuple
+    """
+    return _get_validated(z_util(interfaces.IDisplay).input, validator, *args, **kwargs)
+
+
+def validated_directory(validator, *args, **kwargs):
+    """Like ~certbot.interfaces.IDisplay.directory_select, but with validation.
+
+    :param function validator: A method which will be called on the
```

Comment:
Using "callable" for the type is arguably more accurate and also causes a link to be generated to the Python documentation where "function" does not.

Also, just a reminder to update this docstring if you change how `validator` reports failure.